usr/bin/cylc

```diff
@@ -16,54 +16,123 @@
+# value first to avoid overiding task job version selection:
+# - export CYLC_VERSION=${CYLC_VERSION:-8.0.0}
+#
+# INSTALLING CYLC RELEASES WITH CONDA
+#------------------------------------
+# Create environments named cylc-$CYLC_VERSION or cylc-flow-$CYLC_VERSION under
+# a root location such as /opt/cylc:
+#
+# $ conda create -p /opt/cylc/cylc-8.2.0 cylc=8.2.0
 # 
-# 1) If $CYLC_HOME is defined and exists it will be selected. If it does
-#   not exist the command will fail.
+# $ ls /opt/cylc
+# cylc-8.1.0/
+# cylc-8.2.0/
+#
+# Note you can `pip install` cylc-flow from a git clone into a conda
+# environment and then select it by CYLC_VERSION but that will be misleading
+# (the code won't actually be the release version). It is better to use Python
+# venvs and CYLC_HOME for cylc-flow development and testing.
 #
-# 2) If $CYLC_VERSION is defined and exists it will be selected. If it
-#   does not exist the default under $CYLC_HOME_ROOT will be selected.
+# INSTALLING CYLC-FLOW RELEASES WITH PIP FOR DEVELOPMENT AND TESTING
+#-------------------------------------------------------------------
+# Create Python virtual environments in-place in your cylc-flow git
+# clones and select them using CYLC_HOME in your .bashrc.
+#
+# Create a new venv and install cylc-flow:
+#   $ cd <my-cylc-flow-clone>
+#   $ python -m venv venv
+#   $ . venv/bin/activate
+#   $ pip install -e . 
+#
+# Set CYLC_HOME in .bashrc to select this venv (see explanation above):
+#   $ echo "export CYLC_HOME=$PWD/venv" >> .bashrc
+#   $ . .bashrc
+#
+# You can also `pip install` cylc-flow releases under a common root location
+# for CYLC_VERSION based selection, like for conda:
+#
+#   $ CYLC_HOME_ROOT=$HOME/cylc-flow/releases  # (set in this wrapper too)
+#   $ python -m venv $CYLC_HOME_ROOT/cylc-flow-8.0a2
+#   $ . $CYLC_HOME_ROOT/cylc-flow-8.0a2/bin/activate
+#   $ python -m pip install cylc-flow==8.0a2
 # 
-# 3) If $CYLC_HOME and $CYLC_VERSION are both undefined, the default
-#   under $CYLC_HOME_ROOT will be selected.
+# (But there is no need to do this if you have full system releases - i.e. not
+# just cylc-flow - installed via conda anyway.)
  
-#_____________________________
-# NOTES:
-# * Users will typically accept the default or set $CYLC_VERSION, but
-#   they may set $CYLC_HOME to pick up a private cylc installation.
-# * Developers may set $CYLC_HOME to pick up their own repository clone.
-# * Running suites export $CYLC_VERSION to task environments, so that
-#    task messaging commands etc. pick up the compatible cylc version.
-# * Note that $CYLC_ROOT_HOME can be overridden in the environment too.
-
-CYLC_HOME_ROOT=${CYLC_HOME_ROOT:-/opt} # !!! EDIT ME !!!
+##############################!!! EDIT ME !!!##################################
+# Central Cylc conda releases:
+CYLC_HOME_ROOT=${CYLC_HOME_ROOT:-/opt}
+# User Conda releases:
+CYLC_HOME_ROOT_ALT=${CYLC_HOME_ROOT_ALT:-/home/$USER/miniconda3/envs}
```

Comment:
Nice.

Reply:
(Should be `$HOME/miniconda3/envs` actually; I'll fix that).